### PR TITLE
Update default AP SSID & passphrase

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/OS-1182) |

#### Main changes

Updated default access point credentials:
- Hostname is a mix of the hostname and the last 4 characters of the device serial number. eg: `pi-top-1987`
- Passphrase is a random 8 digit number which is set the first time the access point starts.

#### Screenshots (feature, test output, profiling, dev tools etc)

N/A

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)

N/A

#### Manual testing tips

- Download deb files and scp them into your pi-top
- ssh into your pi-top and install the deb files
- Check the current `hostapd` configuration.
```
pi@pi-top:~ $ cat /etc/hostapd/hostapd.conf
auth_algs=1
channel=6
country_code=00
ctrl_interface=/var/run/hostapd
ctrl_interface_group=0
hw_mode=g
ignore_broadcast_ssid=0
macaddr_acl=0
rsn_pairwise=CCMP
wpa=2
wpa_pairwise=TKIP
interface=wlan_ap0
ssid=pi-top
wpa_passphrase=pi-top1234
```
- Re-configure the AP. Run `sudo wifi-ap-sta configure`.
- Check the `hostapd` configuration again. You should be able to see the new `SSID` & `wpa_passphrase`.
```pi@pi-top:~ $ cat /etc/hostapd/hostapd.conf
auth_algs=1
channel=6
country_code=00
ctrl_interface=/var/run/hostapd
ctrl_interface_group=0
hw_mode=g
ignore_broadcast_ssid=0
macaddr_acl=0
rsn_pairwise=CCMP
wpa=2
wpa_pairwise=TKIP
interface=wlan_ap0
ssid=pi-top-E89E
wpa_passphrase=80691386
```
- This command only updated the configuration files but the AP that's running doesn't know about this. Run `sudo wifi-ap-sta start` to restart AP mode; note that your ssh might hang.
- Try to connect to the AP with the new credentials.
- Verify that the file `/etc/default/wifi-ap-sta` contains the password. This file is used to persist the password on reboot.
```
pi@pi-top:~ $ cat /etc/default/wifi-ap-sta
AP_PASSWORD=80691386
```
- Also, you can try rebooting your device and verifying that the `wpa_passphrase`/`AP_PASSWORD` hasn't changed in `hostapd.conf` & `/etc/default/wifi-ap-sta`.

#### Tag anyone who definitely needs to review or help

N/A

### CODE REVIEW

File: wifi-ap-sta [hunk 1]
```diff
@@ -42,6 +43,23 @@ initialise_ap_values() {
 	HOSTAPD_country_code="${HOSTAPD_country_code:-$(wifi_country)}"
 }
 
+produce_ssid() {
+	local SSID;
+	if [ -f "/proc/cpuinfo" ]; then
+		SSID=$(grep "Serial" /proc/cpuinfo | awk '{print $3}')
+	fi
+	SSID="${SSID:-12345678}"
```
Comment (angusjfw) [inline wifi-ap-sta:51]: this default should be realistic eg 4 chars

**Comment by angusjfw** [comment]
I installed this and rebooted to do onboarding and it worked as expected with new ssid & password.

The only thing that seems a bit off is that after onboarding I checked the config and `/etc/default/wifi-ap-sta` showed the new password but `/etc/hostapd/hostapd.conf` showed the old default still. I turned the AP back on from the miniscreen settings and after that hostapd.conf showed the new password, but it surprised me that config file was not updated on the previous boot.

File: wifi-ap-sta (hunk 1 above)
Comment (jcapona) [inline wifi-ap-sta:51]:
```suggestion
	SSID="${SSID:-1234}"
```

**Comment by jcapona** [comment]
> I installed this and rebooted to do onboarding and it worked as expected with new ssid & password.
> 
> The only thing that seems a bit off is that after onboarding I checked the config and `/etc/default/wifi-ap-sta` showed the new password but `/etc/hostapd/hostapd.conf` showed the old default still. I turned the AP back on from the miniscreen settings and after that hostapd.conf showed the new password, but it surprised me that config file was not updated on the previous boot.

When the AP configuration is performed on boot (if AP mode is enabled), we run `wifi-ap-sta configure`. If a `hostapd.conf` file existed by this time, it will move the file to `hostapd.conf.orig` in order to restore it later, and then write our configuration in the file.

Also, when AP mode is disabled, we run a cleanup step; it will remove custom configurations and restore overwritten files (check [here](https://github.com/pi-top/Wi-Fi-Access-Point-and-Station-Mode/blob/master/wifi-ap-sta#L241)). 

The file you see after reboot is a file that was restored when disabling AP mode. I think you're right, that file shouldn't be there but we could fix that in a separate PR.

File: wifi-ap-sta [hunk 2]
```diff
@@ -42,6 +43,23 @@ initialise_ap_values() {
...
+	SSID=$(echo "${SSID: -4}" | tr '[:lower:]' '[:upper:]')
+	echo "$(hostname)-${SSID}"
+}
+
+produce_password() {
+	local PASSPHRASE;
+	PASSPHRASE="$(shuf -i 10000000-99999999 -n 1)"
```
[the remark this answers is not in the document]
Comment (jcapona) [reply wifi-ap-sta]: updated it to use `/dev/urandom` which is definitely better in terms of generating a secure passphrase...

File: wifi-ap-sta [hunk 3]
```diff
@@ -42,6 +43,23 @@ initialise_ap_values() {
...
+		SSID=$(grep "Serial" /proc/cpuinfo | awk '{print $3}')
+	fi
+	SSID="${SSID:-1234}"
+	SSID=$(echo "${SSID: -4}" | tr '[:lower:]' '[:upper:]')
+	echo "$(hostname)-${SSID}"
+}
+
+produce_password() {
+	local PASSPHRASE;
+	PASSPHRASE="$(< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c8)"
```
Comment (angusjfw) [inline wifi-ap-sta:58]: this includes letters as well as _ and -
I think sticking with numbers is good for convenience - there's less potential for confusion over similar characters
```suggestion
	PASSPHRASE="$(< /dev/urandom tr -dc 0-9 | head -c8)"
```

**Review by angusjfw — CHANGES_REQUESTED** [review]

File: wifi-ap-sta (hunk 3 above)
Comment (angusjfw) [reply wifi-ap-sta:58]: if you feel it's worth including letters for security, at least leave out the punctuation 'A-Za-z0-9'

Comment (jcapona) [inline wifi-ap-sta:58]:
```suggestion
	PASSPHRASE="$(< /dev/urandom tr -dc a-km-z0-9 | head -c8)"
```

**Review by angusjfw — APPROVED** [review]
